### PR TITLE
Add OffensEval-TR 2020 Dataset

### PR DESCRIPTION
**Pull request by yavuzKomecoglu**
This PR adds the OffensEval-TR 2020 dataset which is a Turkish offensive language corpus by me and @basakbuluz. The corpus consist of randomly sampled tweets and annotated in a similar way to [OffensEval](https://sites.google.com/site/offensevalsharedtask/) and [GermEval](https://projects.fzai.h-da.de/iggsa/).

- **Homepage:** [offensive-turkish](https://coltekin.github.io/offensive-turkish/)
- **Paper:** [A Corpus of Turkish Offensive Language on Social Media](https://coltekin.github.io/offensive-turkish/troff.pdf)
- **Point of Contact:** [Çağrı Çöltekin](ccoltekin@sfs.uni-tuebingen.de)

### CODE REVIEW

**Comment by yavuzKomecoglu** [comment]
@lhoestq, can you please review this PR?

File: datasets/offenseval2020_tr/README.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- 1k<n<40k
+source_datasets:
+- original
+task_categories:
+- text_classification
+task_ids:
+- offensive-language-classification
```
Comment (lhoestq) [inline datasets/offenseval2020_tr/README.md:19]:
```suggestion
- text-classification-other-offensive-language-classification
```

File: datasets/offenseval2020_tr/README.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+language_creators:
+- found
+languages:
+- tr
+licenses:
+- found
+multilinguality:
+- monolingual
+size_categories:
+- 1k<n<40k
```
Comment (lhoestq) [inline datasets/offenseval2020_tr/README.md:13]:
```suggestion
- 10k<n<100k
```

File: datasets/offenseval2020_tr/offenseval2020_tr.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    def __init__(self, **kwargs):
+        """BuilderConfig for OffensEval2020TR.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(OffensEval2020TRConfig, self).__init__(**kwargs)
+
+
+class OffensEval2020TR(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/offenseval2020_tr/offenseval2020_tr.py:48]:
```suggestion
class Offenseval2020TR(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `offenseval2020_tr`

File: datasets/offenseval2020_tr/dataset_infos.json [hunk 4]
```diff
@@ -0,0 +1 @@
... = {6174--6184},\n address  = {Marseille, France},\n url  = {https://www.aclweb.org/anthology/2020.lrec-1.758},\n}\n", "homepage": "https://coltekin.github.io/offensive-turkish/", "license": "", "features": {"id": {"dtype": "int32", "id": null, "_type": "Value"}, "tweet": {"dtype": "string", "id": null, "_type": "Value"}, "subtask_a": {"num_classes": 2, "names": ["NOT", "OFF"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "post_processed": null, "supervised_keys": null, "builder_name": "offens_eval2020_tr", "config_name": "offenseval2020-turkish", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 4260505, "num_examples": 31756, "dataset_name": "offens_eval2020_tr"}, "test": {"name": "test", "num_bytes": 481300, "num_examples": 3528, "dataset_name": "offens_eval2020_tr"}}, "download_checksums": {"https://coltekin.github.io/offensive-turkish/offenseval2020-turkish.zip": {"num_bytes": 2048258, "checksum": "7977e96255dbc9b8d14893f1b14cbe3dec53c70358503c062c5a59720ec9c2f2"}}, "download_size": 2048258, "post_processing_size": null, "dataset_size": 4741805, "size_in_bytes": 6790063}}
```
Comment (lhoestq) [inline datasets/offenseval2020_tr/dataset_infos.json:1]:
```suggestion
{"offenseval2020-turkish": {"description": "OffensEval-TR 2020 is a Turkish offensive language corpus. The corpus consist of randomly sampled tweets, and annotated in a similar way to OffensEval and GermEval.\n", "citation": "@InProceedings{coltekin2020lrec,\n author  = {Cagri Coltekin},\n year  = {2020},\n title  = {A Corpus of Turkish Offensive Language on Social Media},\n booktitle  = {Proceedings of The 12th Language Resources and Evaluation Conference},\n pages  = {6174--6184},\n address  = {Marseille, France},\n url  = {https://www.aclweb.org/anthology/2020.lrec-1.758},\n}\n", "homepage": "https://coltekin.github.io/offensive-turkish/", "license": "", "features": {"id": {"dtype": "int32", "id": null, "_type": "Value"}, "tweet": {"dtype": "string", "id": null, "_type": "Value"}, "subtask_a": {"num_classes": 2, "names": ["NOT", "OFF"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "post_processed": null, "supervised_keys": null, "builder_name": "offenseval2020_tr", "config_name": "offenseval2020-turkish", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 4260505, "num_examples": 31756, "dataset_name": "offenseval2020_tr"}, "test": {"name": "test", "num_bytes": 481300, "num_examples": 3528, "dataset_name": "offenseval2020_tr"}}, "download_checksums": {"https://coltekin.github.io/offensive-turkish/offenseval2020-turkish.zip": {"num_bytes": 2048258, "checksum": "7977e96255dbc9b8d14893f1b14cbe3dec53c70358503c062c5a59720ec9c2f2"}}, "download_size": 2048258, "post_processing_size": null, "dataset_size": 4741805, "size_in_bytes": 6790063}}
```

`offens_eval2020_tr`-> `offenseval2020_tr`

**Review by lhoestq — APPROVED** [review]
Awesome thank you !

**Comment by yavuzKomecoglu** [comment]
> Awesome thank you !

Thanks for the small fixes @lhoestq

**Comment by yavuzKomecoglu** [comment]
@coltekin, we have added the data set that you created an article that says "Turkish Attack Language Community in Social Media", HuggingFace dataset update sprint for you. We added Sprint quickly for a short time. I hope you welcome it too. The dataset is accessible at https://huggingface.co/datasets/offenseval2020_tr.

**Comment by coltekin** [comment]
Thank you for the heads up. I am not familiar with the terminology above (no idea what a sprint is), but I am happy that you found the data useful. Please feel free to distribute/use it as you see fit.

The OffensEval version you included in your data set has only binary labels. There is also a version [here](https://coltekin.github.io/offensive-turkish/troff-v1.0.tsv.gz) which also includes fine-grained labels similar to the OffensEval English data set - Just in case it would be of interest.

If you have questions about the data set, or need more information please let me know.

**Pull request merged**